bpf/lib/lb.h

```diff
@@ -628,24 +630,37 @@ lb6_select_backend_id(struct __ctx_buff *ctx,
 #elif LB_SELECTION == LB_SELECTION_MAGLEV
 static __always_inline __u16
 lb6_select_backend_id(struct __ctx_buff *ctx __maybe_unused,
-		      struct lb6_key *key __maybe_unused,
+		      const struct lb6_key *key __maybe_unused,
 		      const struct ipv6_ct_tuple *tuple,
 		      const struct lb6_service *svc)
 {
-	__u32 zero = 0, index = svc->rev_nat_index;
+	__u32 index = svc->rev_nat_index, slot = 0, offset = 0,
+	      lut_size = LB_MAGLEV_LUT_SIZE;
 	__u16 *backend_ids;
 	void *maglev_lut;
 
 	maglev_lut = map_lookup_elem(&LB6_MAGLEV_MAP_OUTER, &index);
 	if (unlikely(!maglev_lut))
 		return 0;
 
-	backend_ids = map_lookup_elem(maglev_lut, &zero);
+	backend_ids = map_lookup_elem(maglev_lut, &slot);
 	if (unlikely(!backend_ids))
 		return 0;
 
-	index = hash_from_tuple_v6(tuple) % LB_MAGLEV_LUT_SIZE;
-	return map_array_get_16(backend_ids, index, (LB_MAGLEV_LUT_SIZE - 1) << 1);
+	/* TODO(christarazi): is this unlikely? */
```

Comment:
I'm fine with the comment - Just double-checking that this is not something you planned to remove before pushing/merging the PR.

Reply:
I put this commit to solicit feedback on whether it's needed.

Reply:
Right, sorry. We have
```
	__u16 backend_slot;	/* Backend iterator, 0 indicates the svc frontend */
```
so I suppose we can't really assume there will be fewer packets for the service frontend than for the backend? Or does it differ with Maglev? (Sorry I don't feel really useful on that one)

Reply:
I'm not sure I understand why this is not done unconditionally? `lut_size = svc->backend_id;`